source/ui/resolution.js

```diff
@@ -0,0 +1,194 @@
+			type = type || _screenType;
+			if (type) {
+				_riRatio = getScreenTypeObject(type).pxPerRem;
+				return _riRatio;
+			}
+			return 1;
+		},
+
+		/**
+		* Takes a provided pixel value and preforms a scaling operation on the number based on the
+		* current screen type.
+		*
+		* @public
+		*/
+		scale: function (px) {
+			return (_riRatio || this.getRiRatio()) * px;
+		},
+
+		/**
+		* The default configurable [options]{@link enyo.ri.selectSrc#options}.
+		*
+		* @typedef {Object} enyo.ri.selectSrc~src
+		* @property {String} hd - HD / 720p Resolution image asset source URI/URL
+		* @property {String} fhd - FHD / 1080p Resolution image asset source URI/URL
+		* @property {String} uhd - UHD / 4K Resolution image asset source URI/URL
+		*
+		* @typedef {String} enyo.ri.selectSrc~src - Image asset source URI/URL
+		*/
+
+		/**
+		* Image src chooser. A simple utility method to select the ideal image asset from a set of
+		* assets, based on various screen resolutions: HD (720p), FHD (1080p), UHD (4k). When provided
+		* with a src argument, multiResSrc will choose the best image with respect to the current screen
+		* resolution. `src` may be either the traditional string, which will pass straight through, or a
+		* hash/object of screen types and their asset sources (keys:screen and values:src). The image
+		* sources will be used chosen when the screen resolution is less than or equal to the provided
+		* screen types.
+		*
+		* ```
+		* // Take advantage of the multi-rez mode
+		* {kind: 'moon.Image', src: {
+		* 	'hd': 'http://lorempixel.com/64/64/city/1/',
+		* 	'fhd': 'http://lorempixel.com/128/128/city/1/',
+		* 	'uhd': 'http://lorempixel.com/256/256/city/1/'
+		* }, alt: 'Multi-rez'},
+		* // Standard string `src`
+		* {kind: 'moon.Image', src: http://lorempixel.com/128/128/city/1/', alt: 'Large'},
+		* ```
+		*
+		* @param {(String|moon.ri.selectSrc~src)} src A string containing a single image src or a
+		*	key/value hash/object containing keys representing screen types (hd, fhd, uhd, etc) and
+		*	values containing the asset src for that target screen resolution.
+		* @returns {String} The choosen src given the string or list provided.
+		* @public
+		*/
+		selectSrc: function (src) {
+			if (typeof src != 'string' && src) {
+				var i, t,
+					newSrc = src.fhd || src.uhd || src.hd,
+					types = _screenTypes;
+
+				// loop through resolutions
+				for (i = types.length - 1; i >= 0; i--) {
+					t = types[i].name;
+					if (_screenType == t && src[t]) newSrc = src[t];
+				}
+
+				src = newSrc;
+			}
+			return src;
+		},
+
+		/**
+		* This will need to be re-run any time the screen size changes, so all the values can be
+		* re-cached.
+		*
+		* @public
+		*/
+		// Later we can wire this up to a screen resize event so it doesn't need to be called manually.
+		init: function () {
+			_screenType = this.getScreenType();
+			this.updateScreenTypeOnBody();
+			enyo.dom.unitToPixelFactors.rem = scope.enyo.getUnitToPixelFactors();
```

Comment:
Ah ha! Maybe I should have started with reviewing 958 instead - was trying to do them discretely to make the review more tractable. :smiley: